src/snowflake/connector/result_batch.py

```diff
@@ -453,11 +457,16 @@ def _parse(
                 except Exception as error:
                     msg = f"Failed to convert: field {col.name}: {_t}::{v}, Error: {error}"
                     logger.exception(msg)
-                    yield Error.errorhandler_make_exception(
-                        InterfaceError,
-                        {"msg": msg, "errno": ER_FAILED_TO_CONVERT_ROW_TO_PYTHON_TYPE},
+                    result_list.append(
+                        Error.errorhandler_make_exception(
+                            InterfaceError,
+                            {
+                                "msg": msg,
+                                "errno": ER_FAILED_TO_CONVERT_ROW_TO_PYTHON_TYPE,
+                            },
+                        )
```

Comment:
I need to double check the behaviour when there is an error (since it seems like an exception and the row_result would be added to the result_list).
There are no tests so I'll see what it returns and go from there.